src/checkInput.js

```diff
@@ -112,7 +111,7 @@ export function checkInput(data, peaks, options) {
     peaks[i].y /= maxY;
   }
 
-  let parameters = assignDeep({}, defaultParameters, optimization.parameters);
+  let parameters = { ...defaultParameters, ...optimization.parameters };
```

Comment:
destructuring assignment does not go deep unlike assignDeep.

Reply:
you are right, it is a wrong way to solve the problem. For some reason, assign-deep doesn't overwrite a property which value is a function. I wrote the next function that makes the correct assignment for the current case.

```javascript
function assignDeep(arrayObjects) {
  let data = arrayObjects.reverse();
  if (data.length === 2) {
    for (let key in data[0]) {
      if (!data[1][key]) {
        data[1][key] = data[0][key];
      } else {
        if (
          typeof data[1][key] === 'object' &&
          typeof data[0][key] === 'object'
        ) {
          data[1][key] = assignDeep([data[1][key], data[0][key]]);
        } else {
          data[1][key] = data[0][key];
        }
      }
    }
    return data[1];
  } else {
    for (let i = 1; i < data.length; i++) {
      data[i] = assignDeep([data[i], data[i - 1]]);
    }
  }
  return data[data.length - 1];
}
```
looking in the github repo of assign-deep I see that the function is similar and the error coming from 2017. I could refactor the function in order to accept the same input as assign-deep.

Reply:
A better-writen function is 

```javascritpt
function assignDeep(target, ...args) {
  let index = 0;
  if (isPrimitive(target)) target = args[index++];
  if (!target) target = {};
  for (; index < args.length; index++) {
    if (!isObject(args[index])) continue;
    for (const key in args[index]) {
      if (isObject(target[key]) && isObject(args[index][key])) {
        assignDeep(target[key], args[index][key]);
      } else {
        target[key] = args[index][key];
      }
    }
    assignSymbols(target, args[index]);
  }
  return target;
}
```
isObject and isPrimitive is
```javascritpt
function isObject(val) {
  return typeof val === 'object';
}
function isPrimitive(val) {
  return typeof val === 'object' ? val === null : typeof val !== 'function';
}
```